ga4gh/datamodel/sequenceAnnotations.py

```diff
@@ -0,0 +1,415 @@
+        term.term, term.id = ontologyTuple[0], ontologyTuple[1]
+        term.sourceName = "sequenceOntology"
+        term.sourceVersion = "0"
+        return term
+
+    def _generateSimulatedFeature(self, randomNumberGenerator):
+        feature = protocol.Feature()
+        # ID and parentId are not generated here.
+        feature.featureSetId = self.getId()
+        feature.start = randomNumberGenerator.randint(1000, 2000)
+        feature.end = feature.start + randomNumberGenerator.randint(1, 100)
+        feature.featureType = self._getRandomOntologyTerm(
+            randomNumberGenerator)
+        references = ["chr1", "chr2", "chrX"]
+        feature.referenceName = randomNumberGenerator.choice(references)
+        strands = [protocol.Strand.POS_STRAND, protocol.Strand.NEG_STRAND]
+        feature.strand = randomNumberGenerator.choice(strands)
+        feature.attributes = protocol.Attributes()
+        feature.attributes.vals = {"gene_name": ["Frances", ],
+                                   "gene_type": ["mRNA", ],
+                                   "gene_status": ["UNKNOWN", ]}
+        return feature
+
+    def getFeature(self, compoundId):
+        """
+        Fetches a simulated feature by ID.
+        :param compoundId: any non-null string
+        :return: A simulated feature with id set to the same value as the
+            passed-in compoundId.
+        ":raises: exceptions.ObjectWithIdNotFoundException if None is passed
+            in for the compoundId.
+        """
+        if compoundId is None:
+            raise exceptions.ObjectWithIdNotFoundException(compoundId)
+        randomNumberGenerator = random.Random()
+        randomNumberGenerator.seed(self._randomSeed)
+        feature = self._generateSimulatedFeature(randomNumberGenerator)
+        feature.id = str(compoundId)
+        feature.parentId = ""  # TODO: Test with nonempty parentIDs?
+        return feature
+
+    def getFeatures(self, referenceName, start, end,
+                    pageToken, pageSize,
+                    featureTypes=[], parentId=None):
+        """
+        Returns a set number of simulated features.
+        :param referenceName: name of reference to "search" on
+        :param start: start coordinate of query
+        :param end: end coordinate of query
+        :param pageToken: None or int
+        :param pageSize: None or int
+        :param ontologyTerms: optional list of ontology terms to limit query
+        :param parentId: optional parentId to limit query.
+        :return: Yields feature, nextPageToken pairs.
+            nextPageToken is None if last feature was yielded.
+
+        """
+        randomNumberGenerator = random.Random()
+        randomNumberGenerator.seed(self._randomSeed)
+        numFeatures = 10
+        if pageToken is not None:
+            nextPageToken = int(pageToken)
+        else:
+            nextPageToken = 0
+        for featureId in ["feature{}".format(x) for x in xrange(numFeatures)]:
```

Comment:
range is much slower than xrange in python 2.  2to3 will handle
the conversion.   If the past is a prior, python 3 is still years
away.  

Jerome Kelleher <notifications@github.com> writes:
> > +        for featureId in ["feature{}".format(x) for x in xrange(numFeatures)]:
> 
> xrange is python 2 only. Use range.
> 
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly or view it on GitHub*
>